### PR TITLE
"Why Mixxx will not support streaming services" blog post

### PR DESCRIPTION
**Pull request by Be-ing**
I hope this puts this perennial topic to rest.

### CODE REVIEW

**Review by daschuer — CHANGES_REQUESTED** [review]
This is not the consolidated opinion of the community, and you should know it.

Instead if putting the discussion to a rest, this PR adds fule to the fire. 

Please close this and not stating the discussion here.

If you desire to building a community opinion, please open a wiki page, collecting pro and cons.

**Comment by Be-ing** [comment]
The purpose of opening this pull request is to work towards a consolidated opinion so we can refer to it instead of continuing to reiterate the same discussion over and over. I will not close it just because one person asks for that. Are there specific parts you disagree with?

**Comment by Holzhaus** [comment]
If we want to discuss that now, can we please discuss this on Zulip instead? I think more people would participate in that discussion. I also find discussions on Zulip much easier to follow when there a lot of replies.

**Comment by Be-ing** [comment]
That discussion has already occurred multiple times and I don't think anyone would say anything new.
https://bugs.launchpad.net/mixxx/+bug/938180
https://mixxx.zulipchat.com/#narrow/stream/109122-general/topic/Soundcloud.20integration

**Comment by Be-ing** [comment]
Even if you are not enthusiastic about everything in this post, this post covers many different reasons and I hope at least some of them are acceptable to everyone.

**Comment by daschuer** [comment]
I have no interest to argue this now.
We have some more interesting topics to argue and I think all arguments are already exchanged. 

The is no that easy: streaming yes or no.

Prominent stating a decision like that about a unknown solution is not a good way to go.
So let's close this and do the discussion along a proposed solution, wichbis pretty much welcome.

My general opinion is unchanged:

We nee streaming, else Mixxx will fall into Oblivion. This already has happened among my friends ...

**Comment by Be-ing** [comment]
I have no interest to argue about this either and it is a waste of time whenever another newbie brings it up. If we do not publish a statement we can reference, then what do we do? Link them to old discussions and tell them to read the whole thing?

**Comment by Be-ing** [comment]
> about a unknown solution

The post explicitly addresses why there never will be any acceptable solution.

**Comment by Holzhaus** [comment]
I just read through the launchpad issue and I looks like there's a split among Mixxx devs on the question whether it's okay to disable features to accomodate streaming services or not. I don't think we should publish a blogpost that state the opinion of one group of devs as the opinion of the whole mixxx dev team.

However, I think we could do a blogpost that doesn't clearly say "we will never integrate streaming" but which explains a little what problems there are and that we can all agree on:
- Most people will want Spotify, but they state in their TOS that it does not allow remixing, overlapping, etc. which is a blocker for Mixxx. Other DJ software probably have special contracts, but since Mixxx is community driven, there is no company that could make a contract with Spotify even if we wanted. So Spotify is out of the picture.
- Even if Spotify made an exception for Mixxx regarding remixing, they won't make an exeption regarding recording and broadcasting (since they didn't even do that for commercial DJ software),
so Mixxx would need to disable these parts while Spotify is used. 
The same restrictions also apply to almost all other popular streaming services, e.g. SoundCloud.
Whether or not this would be okay is very controversial among Mixxx developers.
- That said, if you find a service that does allow Remixing and does not impose such artificial usage restrictions (possibly Jamendo, their TOS leave room for interpretation so we need to ask them), we'd very much appreciate if your contribute code to make it usable in Mixxx. Nobody stepped up to do it yet - probably because streaming also has other disadvantages: the venue you're playing at might not have a decent internet connection, the streaming service might have server problems or might go out of business, etc. which would be a shame if it happened during a set.

**Comment by Be-ing** [comment]
> if you find a service that does allow Remixing and does not impose such artificial usage restrictions

The post explicitly addresses why this will never exist. Even if it hypothetically did, it would be pointless to add any special integration to Mixxx because anyone could simply use normal music files from that service as has always been possible with Mixxx.

I do not think it would be helpful to publish an ambiguous noncommittal statement.

**Comment by Holzhaus** [comment]
I don't want to close that door. There might be other reasons to use streaming, e. g. disk space constraints, being able to react to track wishes from the audience without leaving Mixxx and so on. If someone cares about this enough to contribute the necessary code I'd be open to it.

**Comment by Be-ing** [comment]
> disk space constraints

This doesn't make sense today with storage being so cheap.

> being able to react to track wishes from the audience without leaving Mixxx

This only works if the streaming service is likely to have random tracks that someone in the audience asks for, which means that they have popular music from big labels, which will certainly never allow recording or broadcasting. I would like to definitively close the door to this so we can stop wasting time with these grasping-at-straws arguments for streaming. There is no point of integrating a streaming service if it doesn't have the music people want.

**Comment by Holzhaus** [comment]
> There is no point of integrating a streaming service if it doesn't have the music people want.

If it doesn't have the music people want, nobody will step up and contribute code to add support for it, so why argue about it? And if someone does contribute, I don't want to gatekeep their taste in music or the hardware they use for DJing (which might not have a card reader, an extra HDD slot or a free USB port).

**Comment by Be-ing** [comment]
> If it doesn't have the music people want, nobody will step up and contribute code to add support for it, so why argue about it?

Yes, that's why I'd like to make a definitive, unambiguous statement of "no" so we can stop wasting time discussing it. There's no point saying "well we might accept a contribution that we know no one will ever care enough to actually implement because it's pointless". All that does is leave the door open for more repetitive discussions.

**Comment by Be-ing** [comment]
I added a short paragraph acknowledging the disagreement. I hope we can all agree that regardless of that, the rest of the reasons together are still enough to definitively say no to streaming. Please make inline comments if you have suggestions for the language.

**Comment by daschuer** [comment]
Please let's close this discussion, and save us from the work dealing with this PR.

I will never agree to say paushal "no" to streaming. Our website is the wrong place to give such a statement even if we finally have a team opinion.

If one pays for a steaming service, he should have the freedom to use it with Mixxx.

**Comment by Be-ing** [comment]
I will never agree to allowing streaming in Mixxx and will fork Mixxx if this happens. I am tired of this topic repeatedly distracting us, which will certainly happen if we don't have a link with a prepared statement we can simply copy and paste and move on.

> Our website is the wrong place to give such a statement even if we finally have a team opinion.

Then what is the right place?

> If one pays for a steaming service, he should have the freedom to use it with Mixxx.

That is just one opinion, and it says nothing about the rest of the reasons explained in the post. I think the logic behind does not hold water. Applying the same logic to another situation, many people say Mixxx should be able to mix videos. A user could even say, I have these videos, or I can stream these videos, therefore I should have the freedom to use them with Mixxx. They could even point out that Serato does it, and Virtual DJ does it, so does Rekordbox too, therefore I should have the freedom to do the same with Mixxx. Someone could even say, I have these old video game ROMs, therefore I should have the freedom to have Mixxx run them in an emulator and mix the audio and video output of them. A user could say I have a TSI file that lets my controller work with Traktor, therefore I should have the freedom to use the TSI file with Mixxx. I think these are all illogical. Just because a program could hypothetically be changed to do something in the future does not mean it would be a wise idea for the developers of the program to do that, and it is absurd to imply entitlement to make them do so. We can make decisions to reject certain features and it violates nobody's freedom in any way. We [just did this last week](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/.22Playahead.20mode.22.20(Sandbox)) and [here is another example](https://github.com/mixxxdj/mixxx/pull/2605). Of course, anyone has the right to fork the code and do it themselves if they really feel that strongly about it.

**Comment by Be-ing** [comment]
> We nee streaming, else Mixxx will fall into Oblivion.

I do not understand how you can see the steady growth of the Mixxx community and the improvements made to Mixxx over the past several years and be concerned that Mixxx might fall into oblivion. I have never been more optimistic about Mixxx's future.

**Comment by Be-ing** [comment]
Ping. These repeated discussions with new people rehashing the exact same arguments are tiring.

**Comment by Holzhaus** [comment]
> > If it doesn't have the music people want, nobody will step up and contribute code to add support for it, so why argue about it?
> 
> Yes, that's why I'd like to make a definitive, unambiguous statement of "no" so we can stop wasting time discussing it. There's no point saying "well we might accept a contribution that we know no one will ever care enough to actually implement because it's pointless".

That's not what I meant. If you're right  and nobody wants that kind of music this also means noone would care enough about it to contribute code, which means that there is absolutely no need to write a blogpost that states why Mixxx will never support it. If you're wrong and someone *does* contribute code for streaming from a Service like Jamendo, I don't see any reason why Mixxx shouldn't support it other than "streaming is risky for live sets" - and honestly, that's a decision that's better left to the user.

> All that does is leave the door open for more repetitive discussions.

That's why I think a blogpost regarding the lack of streaming support can make sense. We can link to it when someone asks why there is no streaming support.

I think it should be a post that elaborates on the various legal and ethical issues we face, but I don't want a posts that flat out states "we will never have streaming" if half of the core devs doesn't agree with that and that even rules out streaming support for services that do not have restrictions regarding broadcasting/streaming.

I totally understand that it can be annoying to discuss streaming over and over again. But as with any other feature request, when a user asks why we don't have streaming you can just answer: "Nobody cared enough about it to implement it yet."

The launchpad issue exists since 2012 and in the past 8 years nobody cared enough about this to write some code that implements it. I don't see why we'd need to make a decision on streaming right now.

**Comment by Be-ing** [comment]
> If you're wrong and someone does contribute code for streaming from a Service like Jamendo, I don't see any reason why Mixxx shouldn't support it other than "streaming is risky for live sets" - and honestly, that's a decision that's better left to the user.

This will almost certainly never exist because it would just be a music store as explained in the blog post. Even if such a service existed, I would be opposed to supporting it in Mixxx because it would be a large maintenance burden for very little practical benefit. We can and do reject features for various reasons. We can do the same here.

> I totally understand that it can be annoying to discuss streaming over and over again. But as with any other feature request, when a user asks why we don't have streaming you can just answer: "Nobody cared enough about it to implement it yet."

I am not okay with that answer because it leaves the door open for more distracting, repetitive discussions.

> I don't see why we'd need to make a decision on streaming right now.

Simply because it keeps coming up. None of the reasons to not implement it have changed or will change.

**Comment by ywwg** [comment]
I don't feel this post represents the consensus of the primary Mixxx maintainers.  I agree with Jan that I don't feel comfortable flatly stating we'll never support streaming services.  "We do not currently have plans" is the phrasing I'd prefer.

**Comment by Be-ing** [comment]
@ywwg why should we leave that door open if there is no reason to believe any of the reasons to not implement streaming will change in the future?

**Comment by ywwg** [comment]
Because I'd love to add streaming support if we can find a legal / open source way to make it work. I disagree with having a blanket policy against the concept.

**Comment by Be-ing** [comment]
I thought the blog post was clear about why this will not occur so I did not want to make the post longer by discussing hypothetical situations that will not happen, but since this has come up repeatedly I added a bit to the post explicitly addressing it.

The concept of streaming is restricting saving the downloaded content so it will never be acceptable. Without this restriction, it is simply downloading a music file.

**Comment by daschuer** [comment]
I don't think it is appropriated to put anything of this topic onto our website.
We should write about shiny new features and not about something that is highly controversy. 

I have no issue to conditional disable streaming and recording while playing tracks from a streaming source, if this fullfils the terms and conditions.

**Comment by Be-ing** [comment]
I can understand the desire to only put positive messages on the blog, but then where should this statement go? The wiki?

**Comment by daschuer** [comment]
You can express you concerns in a wiki page in a contra streaming part.

However it would be better to list possible solutions and argue against them. 
I think we all agree that we don't want to become a piracy app.

**Comment by Be-ing** [comment]
I would be okay posting this on the wiki instead of the blog if https://bugs.launchpad.net/mixxx/+bug/938180 is closed as Won't Fix.

**Comment by Be-ing** [comment]
I rebased to get a preview of this [deployed to Netlify](https://deploy-preview-95--mixxx-website.netlify.app/news/2020-05-22-why-mixxx-will-not-support-streaming/). There are no changes to the content though.

**Comment by ywwg** [comment]
I'd like to put netlify behind a simple password or other access mechanism for draft posts.  This content is not merged and approved by all of Mixxx and it could cause confusion to have this content linked online.

**Comment by Be-ing** [comment]
I do not want to make it harder to preview PRs. That will raise the barrier for entry to contributing to the website. Just don't publicize Netlify preview links.

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html]:
```suggestion
If a user didn't want to do this on their own, it would be easy for anyone to do it for them and host downloads of modified versions of Mixxx that could record and broadcast from streaming services.
```

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:52]:
```suggestion
It would be difficult to distance the official Mixxx from these unofficial modifications, which could  lead to negative publicity.
```

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:60]:
```suggestion
Supporting streaming services in Mixxx would require a lot of work to implement.
```

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:65]:
```suggestion
All this hard work could be in vain if the streaming company or a record company they get music from would suddenly revoke access for Mixxx or go out of business.
```

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:67]:
```suggestion
This is not hypothetical - these concerns have already come to pass several times in the short history of DJ software using streaming services.
```

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [inline pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:75]:
```suggestion
Furthermore, you could lose access to the streaming service at the whim of the streaming or record company as discussed above, or if you simply don't renew your subscription - this could even happen during a gig without any warning!
```

**Review by xeruf — COMMENTED** [review]
Found some things while reading through it.
I can understand your frustration @Be-ing, but I also agree with @Holzhaus that you never now what the future might bring, e.g. how everything is moving to the cloud (won't judge that here), we shouldn't give a definitive no.

I also think that this is better fit to the wiki, then we have it as a reference without the strong commitment that a website post has.

**Comment by ywwg** [comment]
> Just don't publicize Netlify preview links.

I'd feel a lot better if we at least had a robots.txt so these things don't start showing up in google

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (Be-ing) [reply pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:65]: I prefer the current wording. I think combining these into one sentence is putting too much into one sentence.

**Comment by daschuer** [comment]
Let's not merge this. It is still to controversial.

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (xeruf) [reply pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:65]: That's a point, but I feel like right now there is too much duplication between them.

**Comment by Holzhaus** [comment]
> > Just don't publicize Netlify preview links.
> 
> I'd feel a lot better if we at least had a robots.txt so these things don't start showing up in google

AFAIK Netlify sets the X-ROBOTS-TAG header on deploy previews, so it shouldn't be indexed by Google. Our custom staging.mixxx.org preview could be indexed though (or do we set that header as well?)

**Comment by Pegasus-RPG** [comment]
Given the level of controversy over this feature, I also am against making a blog post that takes such a hard line. This belongs on the wiki or in the forums.
Besides, I eventually plan to code up a feature to purchase and download tracks from various music stores within Mixxx, mostly because I want that ability. Extending that to support streaming would be trivial AFAICT since the mechanics would largely be the same. (The key difference is that a streamed track is not saved to disk.)

File: pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html
Comment (Pegasus-RPG) [reply pages/news/2020-05-22-why-mixxx-will-not-support-streaming.html:52]: We already have a bigger problem distancing Mixxx from the unofficial versions bundled with spyware and/or sold on eBay.

**Comment by Be-ing** [comment]
Okay, consensus seems to be against posting this on the blog. As I said [before](https://github.com/mixxxdj/website/pull/95#issuecomment-634300434), I'm okay with putting this on the wiki instead if https://bugs.launchpad.net/mixxx/+bug/938180 is closed as Won't Fix.

**Comment by ywwg** [comment]
@Be-ing , I know you feel very strongly about this issue.  The choices you propose seem equivalent -- either we make an official blog post promising never to support streaming, or we close the Streaming Integration bug as "WONT_FIX", i.e. we will not support streaming. This seems like a false choice to me.

You raise a number of good points, in particular around disabling features to comply with license terms and preventing unauthorized hacked versions of Mixxx from being distributed. But judging by the replies from the other people on this thread, there is interest in exploring streaming integration in a way that is consistent with our open source values. There is a long history of gray areas in open source as it intersects with proprietary systems and it sounds like we are not ready to preclude exploration of this area.

**Comment by daschuer** [comment]
> putting this on the wiki instead

Please Don't do this. It is only your opinion. It doesn't make a difference if one find this test on our website or on our wiki.

**Comment by Be-ing** [comment]
I'm not okay with leaving this discussion open without this somewhere on the Mixxx infrastructure we can easily copy and paste a link to.

**Comment by ywwg** [comment]
do you have a personal blog or website? This seems most appropriate for that sort of venue.  I would be ok putting this on the wiki if we created a set of pages for individual developers' personal opinions.  Kind of like the editorial pages in a newspaper.  I think it's ok as a dev to say "here is my opinion and proposal".  But it needs to be clear that this is not (yet??) the consensus view of the mixxx developers.  A statement like "This is a personal opinion and does not reflect the opinion of the Mixxx project as a whole" should be at the top.

**Comment by Be-ing** [comment]
Why couldn't we put it on the wiki with such a disclaimer?

**Comment by ywwg** [comment]
Yes, as I said I think the wiki is fine as long as we create a separate section for developer personal opinions.

**Comment by Holzhaus** [comment]
Why not a forum post?

**Comment by Be-ing** [comment]
The forum is for user support.

**Comment by ywwg** [comment]
Well the forum is for all discussions... that does seem like the perfect place to have this discussion with the community and see what they think

**Comment by Be-ing** [comment]
The point is to not have the same circular discussion over and over again.

**Comment by Holzhaus** [comment]
> The point is to not have the same circular discussion over and over again.

With a forum post you have link where you can point people when the discussion comes up so you don't have to explain your position over and over - as I understand it this was your primary intent with this PR.

**Comment by daschuer** [comment]
This one is outdated.

**Pull request closed without merging**
